server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -328,88 +328,105 @@ private void ensureRequiredNamesPresent(String repositoryName, @Nullable Reposit
         }
 
         /**
-         * @return an iterator over the snapshot IDs in the given repository which match {@link #snapshotNamePredicate}.
+         * An asynchronous supplier of a {@link SnapshotInfo}.
          */
-        private Iterator<SnapshotId> getSnapshotIdIterator(String repositoryName, @Nullable RepositoryData repositoryData) {
+        private interface AsyncSnapshotInfo {
+
+            /**
+             * @return the {@link SnapshotId} of the {@link SnapshotInfo} to be retrieved.
+             */
+            SnapshotId getSnapshotId();
```

Comment:
There'll eventually be no need for this, once we handle `addSimpleSnapshotInfos` in the same way, but I'm leaving it like this for now to keep the PR focussed.